src/backend/assets/js/main.js

```diff
@@ -112,8 +112,49 @@ const loadVideoAndSlides = function(div, videoURL, slideURL) {
   }
 };
 
-window.handleClientLoad = handleClientLoad;
-window.initClient = initClient;
-window.handleAuthClick = handleAuthClick;
-window.listUpcomingEvents = listUpcomingEvents;
-window.loadVideoAndSlides = loadVideoAndSlides;
+$(document).ready(function () {
+  let filter;
+  //store room filters
+  let room ={
+    dateFilter:'.date-filter'
+  };
+  //store track filters
+  let track ={
+    dateFilter:'.date-filter'
+  };
+  //store schedule filters
+  let schedule={
+    dateFilter:'.day-filter'
+  };
+
+  const noResult=function(filterType,calendarMode) {
+    filter=eval(filterType);
+    var search=false;
+    $('.fossasia-filter').each(function() {
+      if($(this).is(':visible') && $(this).val().length!==0) {
+        search=true;
+      }
+    });
+    var listFilterLength =$(filter.dateFilter+':visible').length;
+    var calendarFilterLength = $('.room:visible').length;
+    if((calendarMode && calendarFilterLength == false) || (!calendarMode &&listFilterLength == false)) {
+      $('#no-results').remove();
+      if(search) {
+        $('.date-list').after("<p style = 'padding-left: 15px; margin-top: 15px; font-weight: bold; color: red' id='no-results'>No matching results found.</p>");
+      } else {
+        $('.date-list').after("<p style = 'padding-left: 15px; margin-top: 15px; font-weight: bold; color: red' id='no-results'>No result found.</p>");
+        }
+    } else {
+       $('#no-results').remove();
+      }
+  };
+
+  window.mainObj= {
+    noResult:  noResult
+  };
+  window.handleClientLoad = handleClientLoad;
```

Comment:
Can you put all these functions in `mainObj` and make the necessary changes that would make the implementation consistent. Also wouldn't it be good if we rename the object to just `main`?

Reply:
@agbilotia1998 I think that there is no need to create a object and i should remove it.